src/web/routes/application/send-code/notify.js

```diff
@@ -6,26 +6,40 @@ const { logger } = require('../../../logger')
 
 const { EMAIL, TEXT } = require('../common/constants')
 
+const SEND_2FA_CODE = 'SEND_2FA_CODE'
+
 // template ids matching the templates in govuk notify.
 const EMAIL_TEMPLATE_ID = '18cf4f69-e1b5-4aa9-bed7-c906d3a59285'
 const SMS_TEMPLATE_ID = 'a770479a-097c-4079-acc2-d452a6f21585'
 
-function logConfirmationCodeEvent (channel, reference) {
-  const data = { eventType: 'SEND_2FA_CODE', timestamp: new Date(), codeChannel: channel, notifyReference: reference }
-  logger.info(JSON.stringify(data))
+function logFailureEvent (initialEvent, failureDescription, exceptionDetail) {
```

Comment:
Seems like this PR is a mix of merges from master and actual code. Could you merge from master directly into the story branch first?

Reply:
Done.